src/WebJobs.Extensions.Http/HttpTriggerAttributeBindingProvider.cs

```diff
@@ -408,6 +408,12 @@ private static object ConvertValueIfNecessary(object value, Type targetType)
                     {
                         value = jObject.ToObject(targetType);
                     }
+                    else if (targetType == typeof(Guid) && value.GetType() == typeof(string))
```

Comment:
Below we're doing Nullable.GetUnderlyingType - do we need this here as well? Does your new code work if the method binds to Guid?

Reply:
That's a really good point -- I swear I had tried out that scenario but evidently not. I fixed that and added test cases to cover nullables like you suggested.